diff --git a/include/openssl/base.h b/include/openssl/base.h
index ebbb1cfed..14c53c84f 100644
--- a/include/openssl/base.h
+++ b/include/openssl/base.h
@@ -278,6 +278,9 @@ typedef struct X509_crl_st X509_CRL;
 typedef struct X509_extension_st X509_EXTENSION;
 typedef struct X509_info_st X509_INFO;
 typedef struct X509_name_entry_st X509_NAME_ENTRY;
+#if defined(OPENSSL_WINDOWS) && defined(X509_NAME)
+#undef X509_NAME
+#endif
 typedef struct X509_name_st X509_NAME;
 typedef struct X509_pubkey_st X509_PUBKEY;
 typedef struct X509_req_st X509_REQ;
